Mathlib/Data/ZMod/Defs.lean

```diff
@@ -195,4 +196,10 @@ instance inhabited (n : ℕ) : Inhabited (ZMod n) :=
   ⟨0⟩
 #align zmod.inhabited ZMod.inhabited
 
+/-- Making `ZMod` a discrete topological space. -/
+instance {d : ℕ} : TopologicalSpace (ZMod d) := ⊥
+attribute [local instance] ZMod.instTopologicalSpaceZMod
```

Comment:
What are those attribute declarations even doing?

Reply:
I am not really sure, in Lean 3, these were local instances which I could call when needed. I wanted to reproduce that.

Reply:
I think these are probably fine as global instances?

Reply:
Agreed. And also you're already declaring them to be global with the previous `instance` declarations. So I think your `attribute [local instance]` are ineffective.

Reply:
What other topology would you put? I would think discrete is the usual one.

Reply:
It's worth checking that no non-defeq diamonds are created here; to ensure that, it might be necessary to define the topology via a case split

Reply:
Why not make it a local instance instead?

Reply:
> because we are then putting an instance of a discrete topology structure on the integers..

I think the integers already have this topology anyway?

Reply:
Indeed, see `instTopologicalSpaceInt`

Reply:
In fact, `Fin n` already has the topology too:

https://github.com/leanprover-community/mathlib4/blob/341dab4127bc6f4797c5905708551a23c0f82d01/Mathlib/Topology/Order.lean#L574-L578

Reply:
So given that `ZMod` is always one of these two types, and both have this topology as a global instance, it's pretty clear that ZMod should also have these instances (but defined to have the right defeq)

Reply:
Does the current definition not have the right defeq?